File: src-tauri/src/commands/metrics.rs
use std::collections::HashMap;

use claudette::metrics::{self, AnalyticsMetrics, DashboardMetrics, WorkspaceMetrics};
use tauri::State;

use crate::state::AppState;

#[tauri::command]
pub async fn get_dashboard_metrics(state: State<'_, AppState>) -> Result<DashboardMetrics, String> {
    metrics::dashboard_metrics(&state.db_path).map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn get_workspace_metrics_batch(
    ids: Vec<String>,
    state: State<'_, AppState>,
) -> Result<HashMap<String, WorkspaceMetrics>, String> {
    metrics::workspace_metrics_batch(&state.db_path, &ids).map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn get_analytics_metrics(state: State<'_, AppState>) -> Result<AnalyticsMetrics, String> {
    metrics::analytics_metrics(&state.db_path).map_err(|e| e.to_string())
}

File: src/ui/src/services/tauri/metrics.ts
import { invoke } from "@tauri-apps/api/core";
import type {
  AnalyticsMetrics,
  DashboardMetrics,
  WorkspaceMetrics,
} from "../../types/metrics";

export function getDashboardMetrics(): Promise<DashboardMetrics> {
  return invoke("get_dashboard_metrics");
}

export function getWorkspaceMetricsBatch(
  ids: string[]
): Promise<Record<string, WorkspaceMetrics>> {
  return invoke("get_workspace_metrics_batch", { ids });
}

export function getAnalyticsMetrics(): Promise<AnalyticsMetrics> {
  return invoke("get_analytics_metrics");
}
